### PR TITLE
Add controlled modalize component

### PR DESCRIPTION
**Pull request by AdityaKhatri**

### CODE REVIEW

File: components/General/Modalize.js [hunk 1]
```diff
@@ -1,133 +1,160 @@
...
+export const controlledModalize = (WrappedButtonComponent) => {
+    const ModalComponent = (props) => {
+        const {
+            disabled,
+            modal,
+            className: classNameFromProps,
+            // eslint-disable-next-line no-unused-vars, @typescript-eslint/no-unused-vars
+            initialShowModal,
+            // eslint-disable-next-line no-unused-vars, @typescript-eslint/no-unused-vars
+            onClose,
```
Comment (tnagorra) [inline components/General/Modalize.js]: onClose is being used

File: components/General/Modalize.js [hunk 2]
```diff
@@ -1,133 +1,160 @@
...
-        setWrappedButtonBCR = () => {
-            const wrappedButton = document.getElementsByClassName(this.wrappedButtonClassName)[0];
+export const controlledModalize = (WrappedButtonComponent) => {
+    const ModalComponent = (props) => {
+        const {
+            disabled,
+            modal,
+            className: classNameFromProps,
+            // eslint-disable-next-line no-unused-vars, @typescript-eslint/no-unused-vars
+            initialShowModal,
```
Comment (tnagorra) [inline components/General/Modalize.js]: why not completely remove this for controlled component?

File: components/General/Modalize.js [hunk 3]
```diff
@@ -1,133 +1,160 @@
...
+            // eslint-disable-next-line no-unused-vars, @typescript-eslint/no-unused-vars
+            onClose,
+            showModal,
+            // eslint-disable-next-line no-unused-vars, @typescript-eslint/no-unused-vars
+            onModalVisibilityChange,
```
Comment (tnagorra) [inline components/General/Modalize.js]: onModalVisibilityChange is also being used

File: components/General/Modalize.js [hunk 4]
```diff
@@ -1,133 +1,160 @@
...
+            ...otherProps
+        } = props;
+
+        const [wrappedButtonClassName] = useState(randomString(16));
```
Comment (tnagorra) [inline components/General/Modalize.js]: Use a function to set default value 
```suggestion
        const [wrappedButtonClassName] = useState(() => randomString(16));
```

**Review by tnagorra — APPROVED** [review]

**Review by tnagorra — APPROVED** [review]

**Review by ShrawanBK — COMMENTED** [review]
if `modal` is required in `controlledWrapperPropTypes`, 
should not controlledWraperDefaultProps have a modal set to either null or a default react element?

**Comment by tnagorra** [comment]
> if `modal` is required in `controlledWrapperPropTypes`,
> should not controlledWraperDefaultProps have a modal set to either null or a default react element?

If I am correctly understanding your comment:
If modal is required, then the user must specify a value. So, a default value should not be required.

**Review by samshara — APPROVED** [review]

**Comment by AdityaKhatri** [comment]
> if `modal` is required in `controlledWrapperPropTypes`,
> should not controlledWraperDefaultProps have a modal set to either null or a default react element?

@ShrawanBK , good question. The propTypes is defined to make sure that the devs use it correctly and also to point out that the component is useless without a defined prop. So, it actually works the other way around.

**Pull request merged**